core/domain/feedback_services.py

```diff
@@ -172,3 +173,94 @@ def get_thread_analytics(exploration_id):
     """
     return feedback_jobs.FeedbackAnalyticsAggregator.get_thread_analytics(
         exploration_id)
+
+def create_suggestion(exploration_id, author_id, exploration_version,
+                      state_name, suggestion_content):
+    """Creates a new SuggestionModel object and the corresponding
+    FeedbackThreadModel object."""
+
+    thread_id = feedback_models.FeedbackThreadModel.generate_new_thread_id(
+        exploration_id)
+    (feedback_models.SuggestionModel
+        .create(exploration_id, thread_id, author_id, exploration_version,
+                state_name, suggestion_content))
+    thread = feedback_models.FeedbackThreadModel.create(
+        exploration_id, thread_id)
+    thread.has_suggestion = True
+    thread.exploration_id = exploration_id
+    thread.state_name = state_name
+    thread.original_author_id = author_id
+    thread.status = feedback_models.STATUS_CHOICES_OPEN
+    thread.subject = "SUGGESTION"
+    thread.put()
+
+
+def get_suggestion(exploration_id, thread_id):
+    return feedback_models.SuggestionModel.get(
+        _get_instance_id(thread_id, exploration_id))
+
+
+def get_suggestion_threads(exploration_id, only_pending_required):
+    """Returns a list of all threads that have a pending suggestion, for a
+    given exploration. Returns an empty list if no such threads exist."""
+
+    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    if only_pending_required == True:
+        pending_suggestion_threads = []
+        for thread in threads:
+            if thread.has_suggestion == True:
+                if feedback_models.SuggestionModel.get(
+                    thread.id).status == feedback_models.SUGGESTION_STATUS_NEW:
+                    pending_suggestion_threads.append(thread)
+        return pending_suggestion_threads
+    else:
+        return threads
+
+	
+def _get_instance_id(thread_id, exploration_id):
+    return '.'.join([exploration_id, thread_id])
+
+	
+def _is_suggestion_valid(thread_id, exploration_id):
+    """Check if the suggestion is still valid. A suggestion is  considered
+    invalid if the exploration version has updated since the suggestion was
+    made."""
+
+    return (feedback_models.SuggestionModel.get(
+        id=_get_instance_id(thread_id, exploration_id)).exploration_version ==
+        exp_models.ExpSummaryModel.get(id=exploration_id).version)
+
+
+def accept_suggestion(editor_user_id, change_list, thread_id, exploration_id):
+    """If the suggestion is valid, accepts it by updating the exploration.
+    Raises an exception if the exploration is not valid."""
+
+    from core.domain import exp_services
+
+    if _is_suggestion_valid(thread_id, exploration_id):
+        commit_message = ("Accepted suggestion: %s" % 
```

Comment:
nit: single quotes in Python.

It would be nice to acknowledge the author, and also allow the inclusion of a creator-specified commit message. The end result might be something like:

```
    Accepted suggestion by shantanu: Fixed an error in the number of cows.
```